draft-ietf-quic-qcram.md

```diff
@@ -248,24 +440,21 @@ entries in the dynamic table SHOULD be avoided.  When one of the oldest entries
 in the table is still actively used for references, the encoder SHOULD emit an
 Indexed-Duplicate representation instead (see {{indexed-duplicate}}).
 
-## Refreshing Entries with Duplication {#indexed-duplicate}
-
-~~~~~~~~~~  drawing
-    0 1 2 3 4 5 6 7
-   +-+-+-+-+-+-+-+-+
-   |0|0|1|Index(5+)|
-   +-+-+-+---------+
-~~~~~~~~~~
-{:#fig-index-with-duplication title="Indexed Header Field with Duplication"}
-
-*Indexed-Duplicates* insert a new entry into the dynamic table which duplicates
-an existing entry. {{RFC7541}} allows duplicate HPACK table entries, that is
-entries that have the same name and value.
+## Blocked Decoding
 
-This replaces the HPACK instruction for Dynamic Table Size Update (see Section
-6.3 of {{RFC7541}}, which is not supported by HTTP over QUIC.
+For header blocks encoded in non-blocking mode, the encoder needs to forego
+indexed representations that refer to vulnerable entries (see
+{{overview-hol-avoidance}}).  An implementation could extend the header table
+entry with a boolean to track vulnerability.  However, the number of entries in
+the table that are vulnerable is likely to be small in practice, much less than
+the total number of entries, so a data tracking only vulnerable
+(un-acknowledged) entries, separate from the main header table, might be more
+space efficient.
 
-# Performance considerations
+To track blocked streams, an ordered map (e.g. multi-map) from `Depends Index`
+values to streams can be used.  Whenever the decoder processes a header block on
+the control stream, it can drain any members of the blocked streams map that now
+have their dependencies satisfied.
```

Comment:
Why this complication?  The encoder only needs to track the number of insertions that have been acknowledged.

It gets more complicated if you want to limit the number of blocked decoders, but I'm not yet convinced that we need that feature.  In that case, you only track header blocks that might be blocked and what they block on.